hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -342,7 +343,7 @@ public TargetTagAssignmentResult toggleTagAssignment(final Collection<String> ta
 
         // all are already assigned -> unassign
         if (alreadyAssignedTargets.size() == allTargets.size()) {
-            alreadyAssignedTargets.forEach(target -> target.getTags().remove(tag));
+            alreadyAssignedTargets.forEach(target -> ((JpaTarget) target).removeTag(tag));
```

Comment:
@michahirsch makes no sense, create a cast, generates a another one
@schabdo nope, that means a lot more casts for the _Management_ services